Ginger/GingerCore/ALM/Qtest/ImportFromQtest.cs

```diff
@@ -0,0 +1,1128 @@
+                    }
+                }
+
+                //Add the BF variables (linked variables)
+                if (busVariables.Keys.Count > 0)
+                {
+                    foreach (KeyValuePair<string, string> var in busVariables)
+                    {
+                        //add as String param
+                        VariableString busVar = new VariableString();
+                        busVar.Name = var.Key;
+                        ((VariableString)busVar).InitialStringValue = var.Value;
+                        busFlow.AddVariable(busVar);
+                    }
+                }
+
+                return busFlow;
+            }
+            catch (Exception ex)
+            {
+                Reporter.ToLog(eLogLevel.ERROR, "Failed to import QC test set and convert it into " + GingerDicser.GetTermResValue(eTermResKey.BusinessFlow), ex);
+                return null;
+            }
+        }
+
+        public static void UpdatedQCTestInBF(ref BusinessFlow busFlow, List<QtestTest> tcsList)
+        {
+            if ((busFlow == null) || (tcsList == null) || (tcsList.Count < 1)) return;
+            Dictionary<string, string> busVariables = new Dictionary<string, string>();
+
+            int startGroupActsIndxInBf = 0;
+            Dictionary<string, int> activityGroupsToRemoveIndexes = new Dictionary<string, int>();
+            foreach (QtestTest tc in tcsList)
+            {
+                var activitiesToRemove = busFlow.Activities.Where(x => x.ActivitiesGroupID == tc.TestName).ToList();
+                foreach (Activity activityToRemove in activitiesToRemove)
+                {
+                    if (startGroupActsIndxInBf < busFlow.Activities.IndexOf(activityToRemove))
+                        startGroupActsIndxInBf = busFlow.Activities.IndexOf(activityToRemove);
+                    busFlow.Activities.Remove(activityToRemove);
+                }
+                var activityGroupsToRemove = busFlow.ActivitiesGroups.Where(x => x.ExternalID2 == tc.TestID).ToList();
+                foreach (ActivitiesGroup activityGroupToRemove in activityGroupsToRemove)
+                {
+                    activityGroupsToRemoveIndexes.Add(activityGroupToRemove.ExternalID2, busFlow.ActivitiesGroups.IndexOf(activityGroupToRemove));
+                }
+                foreach (ActivitiesGroup activityGroupToRemove in activityGroupsToRemove)
+                {
+                    busFlow.ActivitiesGroups.Remove(activityGroupToRemove);
+                }
+            }
+
+            int activityGroupToRemoveIndex;
+            foreach (QtestTest tc in tcsList)
+            {
+                activityGroupsToRemoveIndexes.TryGetValue(tc.TestID, out activityGroupToRemoveIndex);
+
+                //check if the TC is already exist in repository
+                ActivitiesGroup tcActivsGroup = new ActivitiesGroup();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this useless assignment to local variable 'tcActivsGroup'.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)